contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+            "OTCWrapper: invalid notional value"
+        );
+        require(_expiry > block.timestamp, "OTCWrapper: expiry must be in the future");
+
+        latestOrder += 1;
+
+        orders[latestOrder] = Order(
+            _underlying,
+            address(0),
+            _isPut,
+            _strikePrice,
+            _expiry,
+            _premium,
+            _notional,
+            _msgSender(),
+            address(0),
+            0,
+            address(0),
+            block.timestamp
+        );
+
+        emit OrderPlaced(latestOrder, _underlying, _isPut, _strikePrice, _expiry, _premium, _notional, _msgSender());
+    }
+
+    /**
+     * @notice cancels an order
+     * @param _orderID order id
+     */
+    function undoOrder(uint256 _orderID) external {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
+        require(orders[_orderID].buyer == msg.sender, "OTCWrapper: only buyer can undo the order");
+        require(orderStatus[_orderID] == OrderStatus.Pending, "OTCWrapper: can only undo pending orders");
+
+        orderStatus[_orderID] = OrderStatus.Failed;
+
+        emit OrderCancelled(_orderID);
+    }
+
+    /**
+     * @notice executes an order
+     * @dev can only be called by whitelisted market makers
+     *      requires that product and collateral have already been whitelisted beforehand
+     *      ensure collateral naked cap from Controller.sol is high enough for the additional collateral
+     *      ensure setUpperBoundValues and setSpotShock from MarginCalculator.sol have been set up
+     * @param _orderID id of the order
+     * @param _userSignature user permit signature
+     * @param _mmSignature market maker permit signature
+     * @param _premium order premium amount
+     * @param _collateralAsset collateral asset address
+     * @param _collateralAmount collateral amount
+     */
+    function executeOrder(
+        uint256 _orderID,
+        Permit calldata _userSignature,
+        Permit calldata _mmSignature,
+        uint256 _premium,
+        address _collateralAsset,
+        uint256 _collateralAmount
+    ) external nonReentrant {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
+        require(orderStatus[_orderID] == OrderStatus.Pending, "OTCWrapper: only pending orders can be executed");
+        require(isWhitelistedMarketMaker[msg.sender], "OTCWrapper: address not whitelisted to execute");
+        require(_userSignature.amount >= _premium, "OTCWrapper: insufficient amount");
+
+        Order memory order = orders[_orderID];
+
+        require(_userSignature.acct == order.buyer, "OTCWrapper: signer is not the buyer");
+        require(block.timestamp <= order.openedAt.add(fillDeadline), "OTCWrapper: deadline has passed");
+        require(whitelist.isWhitelistedCollateral(_collateralAsset), "OTCWrapper: collateral is not whitelisted");
+
+        uint256 orderFee = (order.notional.mul(fee[order.underlying])).div(10000);
```

Comment:
lets add comment why we are doing 10000. Makes sense, but just for mm's

something like 4bps = 0.04% , then need to divide by 100 again so (4 / 100 ) / 100)